sdks/java/core/src/test/java/org/apache/beam/sdk/io/FileBasedSinkTest.java

```diff
@@ -417,6 +419,89 @@ public void testGenerateOutputFilenamesWithoutExtension() {
     assertEquals(expected, actual);
   }
 
+  @Test
+  public void testBuildTemporaryDirectoryName() {
+    // Tests for local files without the scheme.
+    assertThat(
+        new FileBasedWriteOperation.TemporaryDirectoryBuilder()
+            .apply("/home/output"),
+        Matchers.startsWith("/home/temp-beam-output"));
+    assertThat(
+        new FileBasedWriteOperation.TemporaryDirectoryBuilder()
+            .apply("/home/output/"),
+        Matchers.startsWith("/home/output/temp-beam--"));
+
+    // Tests for Windows OS path in URI format.
```

Comment:
I mean "at Beam HEAD", see https://issues.apache.org/jira/browse/BEAM-1045

This PR removes toPath() and resolve(). It fixed most of BEAM-1045. (still need to fix match())
I updated the PR title to refer to BEAM-1045